corehq/apps/reports/standard/deployments.py

```diff
@@ -664,6 +668,26 @@ class AggregateUserStatusReport(ProjectReport, ProjectReportParametersMixin):
     def decorator_dispatcher(self, request, *args, **kwargs):
         super(AggregateUserStatusReport, self).decorator_dispatcher(request, *args, **kwargs)
 
+    @property
+    @memoized
+    def selected_app_id(self):
+        return self.request_params.get(SelectApplicationFilter.slug, None)
+
+    def _get_histogram_aggregation_for_app(self, field_name):
```

Comment:
Hey @zandre-eng 
Can you help understand what is happening in this function?
For someone not familiar with nested aggregation, filter aggregation and data histogram, this function could use some documentation.

Reply:
As far as I know the `NestedAggregation` class is used to query nested data in ES (_"...this data sits inside a nested ES field"_), `FilterAggregation` is used to filter the specific key-value pair (_" correctly retrieve ... according to the filtered app ID"_) and DateHistogram (which I am not familiar with) I presume is to aggregate the "submission/sync date".

Reply:
(the ES aggregation classes is hard to understand...can't say I have an unshakeable grasp on them)

Reply:
Charl is spot on with his explanation. The `DateHistogram` aggregation is used to group documents by a specific time interval. This allows us to aggregate all submission/sync dates that fall on the same dates and show these in a histogram graph on the UI.

It sounds like there is a bit of confusion around how this function works, so I'll look at adding a doc string to better explain what's going on.

Reply:
I have added a docstring to the function ([8bfe57a](https://github.com/dimagi/commcare-hq/pull/34178/commits/8bfe57a7d4007a3781c04577fb95411a8aa07d5f)). @mkangia @Charl1996 let me know if the docstring explains the function well enough, or if it needs any refactoring.

Reply:
Thanks @zandre-eng 

It is a complicated function. I think the naming added to the challenge a bit.

Reply:
I'm happy to change the function name if it's bringing additional confusion. @mkangia Any suggestions on what naming would be better?

Reply:
Function naming is okay. It was the variable naming that got me which I highlighted in the other comment.